src/Common/Notifications/Controller.php

```diff
@@ -0,0 +1,128 @@
+<?php
+/**
+ * Controller for In-App Notifications.
+ *
+ * @since   TBD
+ *
+ * @package TEC\Common\Notifications
+ */
+
+namespace TEC\Common\Notifications;
+
+use TEC\Common\Contracts\Provider\Controller as Controller_Contract;
+
+/**
+ * Class Controller
+ *
+ * @since   TBD
+
+ * @package TEC\Common\Notifications
+ */
+class Controller extends Controller_Contract {
+
+	/**
+	 * Registers actions and filters.
+	 *
+	 * @since TBD
+	 */
+	public function do_register(): void {
+		$this->add_actions();
+	}
+
+	/**
+	 * Unhooks actions and filters.
+	 */
+	public function unregister(): void {
+		$this->remove_actions();
+	}
+
+	/**
+	 * Add the action hooks.
+	 *
+	 * @since TBD
+	 */
+	public function add_actions() {
+		add_action( 'tribe_plugins_loaded', [ $this, 'boot' ], 50 );
+
+		add_action( 'tec_ian_icon', [ $this, 'show_icon' ] );
+
+		add_action( 'wp_ajax_ian_optin', [ $this, 'opt_in' ] );
+		add_action( 'wp_ajax_ian_get_feed', [ $this, 'get_feed' ] );
+		add_action( 'wp_ajax_ian_dismiss', [ $this, 'handle_dismiss' ] );
+	}
+
+	/**
+	 * Remove the action hooks.
+	 *
+	 * @since TBD
+	 */
+	public function remove_actions() {
+		remove_action( 'tribe_plugins_loaded', [ $this, 'boot' ], 50 );
+
+		remove_action( 'tec_ian_icon', [ $this, 'show_icon' ] );
+
+		remove_action( 'wp_ajax_ian_optin', [ $this, 'opt_in' ] );
+		remove_action( 'wp_ajax_ian_get_feed', [ $this, 'get_feed' ] );
+		remove_action( 'wp_ajax_ian_dismiss', [ $this, 'handle_dismiss' ] );
+	}
+
+	/**
+	 * Hook the IAN initialization.
+	 *
+	 * @since TBD
+	 */
+	public function hook_init(): void {
+		add_action( 'admin_init', [ $this, 'initialize' ], 5 );
+	}
+
+	/**
+	 * Boot our internal IAN code.
+	 *
+	 * @since TBD
+	 *
+	 * @return void
+	 */
+	public function boot() {
+		$this->container->make( Notifications::class );
+	}
```

Comment:
As a result i would rename the method - to avoid confusion with the containers boot method being called during the requests lifecycle - and here only register the assets Notifications provide.

No need for the assets to be registered inside Notifications either way, since Notifications is more a business logic interface.